polaris/dataset/_dataset.py

```diff
@@ -222,7 +222,12 @@ def zarr_root(self):
             if saved_on_hub:
                 self._zarr_root = self.client.open_zarr_file(self.owner, self.name, self.zarr_root_path, "r+")
             else:
-                self._zarr_root = zarr.open_consolidated(self.zarr_root_path, "r+")
+                try:
+                    # Attempt to access .zmetadata to open consolidated file
+                    _ = self.zarr_root.store[".zmetadata"]
+                    self._zarr_root = zarr.open_consolidated(self.zarr_root.store, mode="r")
+                except KeyError:
+                    self._zarr_root = zarr.open(self.zarr_root.store, mode="r+")
```

Comment:
I don't think this will work, because it recurses indefinitely. In `zarr_root()` you call `zarr_root()`. 

I also think I explained myself poorly! :sweat_smile: 

I suggest we _do_ throw an error. That way, if people create a custom Zarr archive themselves and it doesn't work, they get an informative error on what went wrong. I suggest: 

```py
try:
    if saved_on_hub:
        self._zarr_root = self.client.open_zarr_file(self.owner, self.name, self.zarr_root_path, "r+")
    else:
        self._zarr_root = zarr.open_consolidated(self.zarr_root.store, mode="r")
except KeyError as error:
    raise InvalidDatasetError("A Zarr archive associated with a Polaris dataset has to be consolidated.")
```

Reply:
Would explain why all the tests are failing 😅 I will update it with these changes!

Reply:
You will still have to update the test cases I believe because that [Zarr archive](https://github.com/polaris-hub/polaris/blob/main/tests/conftest.py#L78-L84) we use is not consolidated! :slightly_smiling_face: